internal/controller/linstorcluster_controller.go

```diff
@@ -199,6 +199,18 @@ func (r *LinstorClusterReconciler) reconcileAppliedResource(ctx context.Context,
 			return err
 		}
 
+		// Special case for satellites on "NoSchedule" nodes. There is a possible race where we:
+		// * Performed a successful check that the satellite already exists
+		// * LinstorSatellite gets removed
+		// * We only now end up here
+		// We would now re-create the satellite, unless we ensure the UID is already set, in which case
+		// the Patch() will fail. We then retry, but this time we will see that the satellite has been removed.
+		if i := slices.IndexFunc(existingSatellites.Items, func(s piraeusiov1.LinstorSatellite) bool {
```

Comment:
We already check which satellite nodes have an existing satellite above. I feel like it would be cleaner to do this UID propagation there. However, I can't see how that would work. We would need to somehow attach the UID to the satellite Node and then embed it in the ResMap. What do you think?

Reply:
Done that now... I also moved the taint/tolerations logic a bit closer to where the generate the Satellites. Using a debugger, I verified that the UID is set as expected, and that we get the expected error on applying a deleted satellite.